src/app/components/signin/signin.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Messages } from './../../model/messages';
+import { AuthService } from './../../services/auth.service';
+import { Component, OnInit, Output, EventEmitter } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { faUser, faKey } from '@fortawesome/free-solid-svg-icons';
+import { parseErrors } from 'src/app/commons/response-utils';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-signin',
+  templateUrl: './signin.component.html',
+  styleUrls: ['./signin.component.css']
+})
+export class SigninComponent implements OnInit {
+  faUser = faUser;
+  faKey = faKey;
+
+  signInForm: FormGroup;
+  submitted = false;
+
+  @Output()
+  messages = new EventEmitter<Messages>();
+
+  @Output()
+  loading = new EventEmitter<boolean>();
+
+  constructor(private formBuilder: FormBuilder, private authService: AuthService, private router: Router) { }
+
+  ngOnInit() {
+    this.signInForm = this.formBuilder.group({
+      username: ['', Validators.required],
+      password: ['', [Validators.required]],
```

Comment:
why on as single element and one as array?

Reply:
Well thats how reactive form builder accepts params, idk why :/